File: client-app/src/app/models/activity.ts
export interface Activity {
    id: string | undefined
    title: string
    date: string
    description: string
    category: string
    city: string
    venue: string
  }

export interface ActivityFormType {
    title: string
    date: string
    description: string 
    category: string
    city: string
    venue: string
  
}


export const mapFormToActivity = (form: ActivityFormType, id: string | undefined): Activity => {
    return {
      id,
      title : form.title,
      date : form.date,
      description:form.description,
      category : form.category,
      city : form.city,
      venue : form.venue
    }
}

File: Domain/Activity.cs
namespace Domain
{
    public class Activity
    {

        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public bool IsCancelled { get; set; }

        public ICollection<ActivityAttendee> Attendees { get; set; } = [];

        public void EditActivity(Activity activity)
        {
            Title = activity.Title;
            Date = activity.Date;
            Description = activity.Description;
            Category = activity.Category;
            City = activity.City;
            Venue = activity.Venue;
            // IsCancelled = activity.IsCancelled;
        }
    }
}